cmd/kube-apiserver/app/server.go

```diff
@@ -544,12 +545,29 @@ func BuildAuthenticator(s *options.ServerRunOptions, extclient clientgoclientset
 		versionedInformer.Core().V1().Secrets().Lister().Secrets(v1.NamespaceSystem),
 	)
 
+	if EgressSelector != nil {
+		egressDialer, err := EgressSelector.Lookup(egressselector.Master.AsNetworkContext())
+		if err != nil {
+			return nil, nil, err
+		}
+		authenticatorConfig.CustomDial = egressDialer
```

Comment:
Won't this break if there's already a custom dialer? I guess I expected this to make a thingy which dials the proxy, and returns a second thingy with a .Dial() method which can be used to make the actual request, and this latter thingy's Dial() method should be called by the existing CustomDial?

Reply:
Generic webhooks are routable from the master so we're just using the default net Dialer (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go#L35). This doesn't dial the proxy at all.

I don't think there are any cases with existing custom dialers since we created the field with this PR and all generic (non admission) webhooks use the default dialer.